src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -3794,6 +3795,10 @@ def set_up_addon_profiles(self, mc: ManagedCluster) -> ManagedCluster:
                 addon_profile.config[CONST_ACC_SGX_QUOTE_HELPER_ENABLED] = "true"
             addon_profiles[CONST_CONFCOM_ADDON_NAME] = addon_profile
             addons.remove('confcom')
+        if 'open-service-mesh' in addons:
```

Comment:
I would appreciate it if you could add some unit test for this part. But if you want to merge the code before the next release, I think it's ok...

Reply:
Isn't this tested as a part of `test_aks_disable_openservicemesh_addon` in `test_aks_commands.py`?

Reply:
The test cases defined in `test_aks_commands.py` are basically E2E tests and I hope you could add some unit test cases for `set_up_addon_profiles`.

Reply:
@FumingZhang I will add unit tests for these changes later on. This change is time sensitive for Azure, and I hope the e2e tests suffice for now.